homework/list.js

```diff
@@ -1,44 +1,51 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+var ListManager = function() {
+  this._listNoteContent = [];
+  this._wrapper =document.querySelector('#note-list-wrapper');
+};
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+ListManager.prototype = {
+
+  start() {
+      this.fetchList().then((function(response) {
+      this.updateList(response);
+      this.drawList();
+      this.preloadFirstNote();
+    }).bind(this)).catch((function() {  
```

Comment:
Serious issue: don't catch error without logging AND re-throwing it, since this will make the error disappear silently. Consider what will happen if the chain contains an error but your program won't alert anything about that.